From 86a8cd9f6a039889801b5bec865a4bc3deb30f47 Mon Sep 17 00:00:00 2001
From: Sven Strickroth2 <email@cs-ware.de>
Date: Mon, 20 Mar 2017 11:21:00 +0100
Subject: [PATCH] filebuf: fix resolving absolute symlinks

The symlink destination is always concatenated to the original path. Fix
this by using `git_buf_sets` instead of `git_buf_puts`.
---
 src/filebuf.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/filebuf.c b/src/filebuf.c
index ef68b16f4..825b9c04c 100644
--- a/src/filebuf.c
+++ b/src/filebuf.c
@@ -246,7 +246,7 @@ static int resolve_symlink(git_buf *out, const char *path)
 
 		root = git_path_root(target.ptr);
 		if (root >= 0) {
-			if ((error = git_buf_puts(&curpath, target.ptr)) < 0)
+			if ((error = git_buf_sets(&curpath, target.ptr)) < 0)
 				goto cleanup;
 		} else {
 			git_buf dir = GIT_BUF_INIT;
-- 
2.12.2

